src/test/e2e-emulated/contracts/semantics.tact

```diff
@@ -0,0 +1,233 @@
+struct SA {
+    a1: Int;
+    a2: SB;
+}
+
+struct SB {
+    b1: Bool;
+    b2: SC;
+    b3: Int;
+}
+
+struct SC {
+    c1: Int;
+}
+
+fun getNewStruct(s: SA): SA {
+    // Reassign the parameter value
+    // This does not change the struct passed as parameter to the function
+    s = SA {a1: 0, a2: SB {b1: false, b2: SC {c1: 0}, b3: 0}};
+    return s;
+}
+
+fun copyAndModify(s: SA): SA {
+    // Change some fields in the struct given as parameter
+    // Since structs are passed by value, this does not change the struct passed as parameter to the function
+    s.a1 += 100;
+    s.a2.b2.c1 += 1;
+    return s;
+}
+
+extends mutates fun reassignStruct(self: SA) {
+    // Reassigning the parameter will change the struct given as parameter
+    self = SA {a1: 0, a2: SB {b1: false, b2: SC {c1: 0}, b3: 0}};
+}
+
+extends mutates fun changeSomeFields(self: SA) {
+    // Change some fields in the struct given as parameter
+    // This will change the struct passed as parameter to the function
+    self.a1 += 100;
+    self.a2.b2.c1 += 1;
+}
+
+contract SemanticsTester {
+
+    // Currently, declaring fields like this:
+    // 
+    // sC: SC = SC {c1: 5}; 
+    // sB: SB = SB {b1: true, b2: sC, b3: 10};
+    // sA: SA = SA {a1: 20, a2: sB};
+    //
+    // or like this:
+    //
+    // sC: SC = SC {c1: 5}; 
+    // sB: SB = SB {b1: true, b2: self.sC, b3: 10};
+    // sA: SA = SA {a1: 20, a2: self.sB};
+    //
+    // gives a compilation error. So, we need to define the fields like this:
```

Comment:
Please file an issue with a minimized example of structs being initialized in terms of other structs. We can support this in const-eval, as we can treat initialized storage vars as constants at this point.